### PR TITLE
Add helper script 'openqa-label-all'

### PR DESCRIPTION
**Pull request by okurz**

### CODE REVIEW

**Comment by Martchus** [comment]
I miss tracking the dependencies of that script. It is installed and hence will be part of the package but the dependencies are not present in the spec file. On the other side it also feels wrong just to depend on Python libraries just for that script. Maybe add a optional dependencies at least?

**Comment by okurz** [comment]
good point. I assume all dependencies are provided by python but a test is better :) Let's see how they fare in the travis CI environment.

**Comment by okurz** [comment]
hm, from https://travis-ci.org/os-autoinst/openQA/jobs/483417752

```
  File "script/openqa-label-all", line 5, in <module>
    from future.standard_library import install_aliases  # isort:skip to keep 'install_aliases()'
ImportError: No module named future.standard_library
```

is this python2?

**Comment by okurz** [comment]
```
$ docker run -it --entrypoint '' -v $PWD:/opt/openqa -v /var/run/dbus:/var/run/dbus registry.opensuse.org/devel/openqa/containers/openqa_dev python --version
Python 2.7.13
```

yes, it is.

So I see the following options:
1. Use "tox" with python environments within the openQA tests for python components
2. Explicitly call the script with `python3` within the tests although it is compatible with python2 as well
3. Provide the python2 dependencies for the "openqa_dev" container
4. Manage the script in another project -> makes it harder for people to find it and we need the complete development process there as well

something else? For now I would go with 2

**Comment by okurz** [comment]
rebased

File: openQA.spec [hunk 1]
```diff
@@ -177,6 +177,8 @@ Requires:       perl(LWP::UserAgent)
 Requires:       perl(Mojolicious)
 Requires:       perl(Regexp::Common)
 Requires:       perl(Try::Tiny)
+Requires:       python3-future
```
Comment (coolo) [inline openQA.spec]: Please move this into a special subpackage. the openQA client doesn't need python3

**Comment by kraih** [comment]
This can't be a Perl script because?

**Comment by Martchus** [comment]
I wouldn't require writing those extra scripts in a particular language - just avoid very uncommon ones which almost nobody is familiar with (e.g. Haskell). It is also not the first Python script in our repos (see crop.py).

But making it a separate (split) package would be preferable (although the dependency size likely does not matter these days anymore).

File: openQA.spec (hunk 1 above)
Comment (kalikiana) [reply openQA.spec]: So... will there be a separate package for Python utilities then? The last update didn't address this but there wasn't a response to this comment either.

I'm not sure if there was a reason to use Python here over Perl. I love Python, but I second that the dependency isn't very appealing when the client doesn't use any Python.

Comment (okurz) [reply openQA.spec]: fixed, please see the updated changes. Moved into the subpackage "python-scripts" now.

**Comment by Martchus** [comment]
The test failure is actually related:  
```
/usr/bin/env: ‘python3’: No such file or directory
#   Failed test at ./t/40-script_openqa-label-all.t line 27.
#          got: '32512'
#     expected: '0'
# Looks like you failed 1 test of 2.
./t/40-script_openqa-label-all.t .......... 
```

**Comment by okurz** [comment]
thanks, haven't seen a test result that helpful lately ;)

File: docker/openqa/entrypoint.sh [hunk 2]
```diff
@@ -13,6 +13,7 @@ export LC_ALL="en_US.UTF-8"
   sudo zypper --gpg-auto-import-keys -n ref --force && \
   sudo zypper up -l -y
 
+sudo zypper --gpg-auto-import-keys -n ref --force && zypper -n in python3-base python3-future python3-requests
```
Comment (okurz) [inline docker/openqa/entrypoint.sh]: this is just temporary and needs to be removed if succesfull as dependency

**Comment by okurz** [comment]
The test is successfully executed in https://travis-ci.org/os-autoinst/openQA/jobs/554662936#L1110 

-> created https://github.com/os-autoinst/openQA/pull/2160 for dependencies

**Comment by okurz** [comment]
All tests fine except "developer fullstack" which failed again. Let's get this over with.

**Comment by AdamWill** [comment]
A Python script that calls the perl client via subprocess feels kinda awful, but using [my lovely Python client library](https://github.com/os-autoinst/openQA-python-client) would cause a kinda circular dependency, I guess...;)

**Comment by AdamWill** [comment]
Also, on a more concrete note: adding the `%python_script_requires` to `%t_requires` means the `-common` subpackage `Requires:` them. I don't think that's intended, is it?

**Comment by okurz** [comment]
@AdamWill the additional dependency is something I would like to fix with https://github.com/os-autoinst/openQA/pull/2161

**Pull request merged**